src/Nri/Ui/Svg/V1.elm

```diff
@@ -1,42 +1,109 @@
 module Nri.Ui.Svg.V1 exposing
     ( Svg
-    , withColor
+    , withColor, withLabel, withWidth, withHeight
     , fromHtml, toHtml
     )
 
 {-|
 
 @docs Svg
-@docs withColor
+@docs withColor, withLabel, withWidth, withHeight
 @docs fromHtml, toHtml
 
 -}
 
+import Accessibility.Styled.Widget as Widget
 import Css exposing (Color)
 import Html.Styled as Html exposing (Html)
 import Html.Styled.Attributes as Attributes
 
 
-{-| -}
+{-| Opaque type describing a non-interactable Html element.
+-}
 type Svg
-    = Svg (Html Never)
-
-
-{-| -}
-withColor : Color -> Svg -> Svg
-withColor color (Svg svg) =
-    Svg (Html.span [ Attributes.css [ Css.color color ] ] [ svg ])
+    = Svg
+        { icon : Html Never
+        , color : Maybe Color
+        , width : Maybe Css.Px
+        , height : Maybe Css.Px
+        , label : Maybe String
+        }
 
 
 {-| Tag html as being an svg.
 -}
 fromHtml : Html Never -> Svg
-fromHtml =
+fromHtml icon =
     Svg
+        { icon = icon
+        , color = Nothing
+        , height = Nothing
+        , width = Nothing
+        , label = Nothing
+        }
+
+
+{-| -}
+withColor : Color -> Svg -> Svg
+withColor color (Svg record) =
+    Svg { record | color = Just color }
+
+
+{-| -}
+withLabel : String -> Svg -> Svg
+withLabel label (Svg record) =
+    Svg { record | label = Just label }
+
+
+{-| -}
+withWidth : Css.Px -> Svg -> Svg
+withWidth width (Svg record) =
+    Svg { record | width = Just width }
+
+
+{-| -}
+withHeight : Css.Px -> Svg -> Svg
+withHeight height (Svg record) =
+    Svg { record | height = Just height }
```

Comment:
it looks like these set max width and max height according to the aspect ratio of the icon itself—that is, they will not necessarily set the exact width/height. Maybe `setWidthBound` or `setMaxWidth` or something else that indicates exactly what's going on?

Tempting to do…

```elm
type Bounds
    = MaxWidth Int
    | MaxHeight Int
    | MaxRect { width : Int, height : Int }

withBounds : Bounds -> Svg -> Svg
```

Reply:
Our SVGs should always have `Attributes.width "100%", Attributes.height "100%"` as properties, so yeah, this property won't usually change the aspect ratio. I didn't want to do `maxHeight` or `maxWidth` because it might get confusing if you were actually trying to set the max-height or max-width CSS properties.

Reply:
Although, there's nothing enforcing the `Attributes.width "100%", Attributes.height "100%"` properties apart from convention -- we could end up with an Svg element that behaved different